specs/agents/span-links-tmp.md

```diff
@@ -0,0 +1,22 @@
+TODO: this content will be appended to "span-links.md" when that file is added in https://github.com/elastic/apm/pull/600
+
+### API
+
+Agents MAY provide a public API to add span links at span/transaction creation.
+A use-case for user's manually adding span links is for [batch message processing](tracing-instrumentation-messaging.md#batch-message-processing)
+that the APM agent does not or cannot instrument. (For some agents it would be
+a burden to internally support span links and *not* expose the API publicly.)
+
+TODO: For the .NET APM agent is there much value in providing a public API to add span links? Perhaps usage of the Activity API and the OTel Bridge dominates?
```

Comment:
I'd not say that the Activity API dominates... it gets adaption, but it's far from being widely used.

I see the [other command regarding Java](https://github.com/elastic/apm/pull/620#discussion_r833973700) - our .NET API is similar in a sense that long term we expect Activity being the primary API for manual instrumentation. We haven't discussed span links specifically for .NET yet.

It's not a huge effort to add this to the public API, so at this point I'd lean towards adding it to our public API in .NET.